frame/assets/src/lib.rs

```diff
@@ -414,6 +435,7 @@ decl_module! {
 
 				*maybe_details = None;
 				Account::<T>::remove_prefix(&id);
+				Approvals::<T>::remove_prefix(&id);
```

Comment:
This is not okay, you will probably need to redesign the system or remove it.

`remove_prefix` will be a storage operation per user approved, and thus can be incredibly large and not predictable/weighable.

Reply:
Fine to put it behind the trait check? `if Self::ApprovalSpendingAllowed::get()`?

Reply:
Unfortunately this does not address my problem. The problem is that `remove_prefix` in not really that safe of an operation within the runtime and without restrictions.

It is probably better to use  a vector of users of bounded length than a hash map.

If you want to use a hash map, we might want to track somewhere the total number of approved users, and then have the user report to us how many approved users there are, we check it, and we use this for weight.

You should probably keep the approvals removal a completely separate operation with its own extrinsic so it can be measured on its own and stuff.

Also there should be a deposit placed and tracked for each approval placed on chain, since it takes up space. The deposit should be refunded when the approvals are cleaned up.

As you can see, this feature is not so trivial to do when done correctly, which is maybe why it was not included.

With something like Proxies, we probably dont need this feature anyway.

Reply:
Will remove it.